app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/SearchTest.kt

```diff
@@ -0,0 +1,213 @@
+import dagger.hilt.android.testing.BindValue
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.UninstallModules
+import kotlinx.coroutines.runBlocking
+import org.hamcrest.CoreMatchers
+import org.junit.*
+import org.junit.runner.RunWith
+import javax.inject.Inject
+
+@UninstallModules(DatabaseManagementModule::class)
+@HiltAndroidTest
+@RunWith(AndroidJUnit4::class)
+class SearchTest {
+
+    @get:Rule
+    val hiltRule = HiltAndroidRule(this)
+
+
+    @get:Rule
+    val activityScenarioRule: ActivityScenarioRule<MainActivity> = ActivityScenarioRule(
+        Intent(
+            ApplicationProvider.getApplicationContext(),
+            MainActivity::class.java
+        )
+    )
+
+    @BindValue
+    @Demo2Database
+    val dbManagement: DatabaseManagement = DefaultDatabaseManagement(DummyDatabaseService())
+
+    private val dummyDatasets = runBlocking { dbManagement.getDatasets() }
+
+    @Before
+    fun setup() {
+        hiltRule.inject()
+        Intents.init()
+    }
+
+    @After
+    fun cleanUp() {
+        Intents.release()
+        activityScenarioRule.scenario.close()
+    }
+
+    @Test
+    fun datasetsOverviewInputEmptyStringYieldsAllResults() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("    "))
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(dummyDatasets.size)
+            )
+        )
+    }
+
+    @Test
+    fun datasetsOverviewSearchByKeyWordYieldsCorrectResult() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("kitchen"))
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(1)
+            )
+        )
+    }
+
+    @Test
+    fun datasetsOverviewSearchNotFoundYieldsNoResult() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("hi"))
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(0)
+            )
+        )
+    }
+
+    @Test
+    fun datasetsOverviewCanClickOnSubsetOfDatasetsMatchingSearch() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("kitchen"))
+
+        onView(withId(R.id.DatasetList_list))
+            .perform(
+                RecyclerViewActions.actionOnItemAtPosition<DatasetListRecyclerViewAdapter.ListItemViewHolder>(
+                    0,
+                    click()
+                )
+            )
+        assertCurrentFragmentIsCorrect(R.id.displayDatasetFragment)
+    }
+
+    @Test
+    @Ignore // haven't found a way to clear the text
```

Comment:
Is this comment still valid because you seem to be doing just that ie perform(clearText(), typeText("smtg")) ?

Reply:
Yes but it doesn't work with the search bar. I think it's only for editText, and I haven't found an equivalent for this.